### PR TITLE
add shenyu.register.enabled

### PR DESCRIPTION
**Pull request by zzq1314zll**
for issue #3703

### CODE REVIEW

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java [hunk 1]
```diff
@@ -27,12 +27,15 @@
...
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.env.Environment;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 
 /**
  * The type shenyu spring cloud client configuration.
  */
 @Configuration
 @ImportAutoConfiguration(ShenyuClientCommonBeanConfiguration.class)
+@ConditionalOnProperty(value = "shenyu.register.enabled",
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:37]: Hi， mybe modify this ：     @ConditionalOnProperty(name = "shenyu.register.enabled", havingValue = "true", matchIfMissing = true)

and you can complete the registration switch for other clients and update the documentation?

Comment (dragon-zhang) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:37]: Maybe it's better to put it in `shenyu-client-springcloud`? `shenyu-spring-boot-starter-client-springcloud` is only used as the wrapper of `shenyu-client-springcloud`.

Comment (zzq1314zll) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:37]: Why?

Comment (zzq1314zll) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:37]: > Maybe it's better to put it in `shenyu-client-springcloud`? `shenyu-spring-boot-starter-client-springcloud` is only used as the wrapper of `shenyu-client-springcloud`.

I want to know what you think

Comment (yu199195) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:37]: mybe add spring-boot-starter is the right

Comment (dragon-zhang) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:37]: got it

**Pull request merged**